core/bin/external_node/src/config/observability.rs

```diff
@@ -98,4 +104,40 @@ impl ObservabilityENConfig {
         }
         Ok(guard)
     }
+
+    pub(crate) fn from_configs(general_config: &GeneralConfig) -> anyhow::Result<Self> {
+        let (sentry_url, sentry_environment, log_format, log_directives) =
+            if let Some(observability) = general_config.observability.as_ref() {
+                (
+                    observability.sentry_url.clone(),
+                    observability.sentry_environment.clone(),
+                    observability
+                        .log_format
+                        .parse()
+                        .context("Invalid log format")?,
+                    observability.log_directives.clone(),
+                )
+            } else {
+                (None, None, LogFormat::default(), None)
+            };
+        let (prometheus_port, prometheus_pushgateway_url, prometheus_push_interval_ms) =
+            if let Some(api) = general_config.api_config.as_ref() {
+                (
+                    Some(api.prometheus.listener_port),
+                    Some(api.prometheus.pushgateway_url.clone()),
+                    api.prometheus.push_interval_ms.unwrap_or_default(),
```

Comment:
Do I understand correctly that `prometheus` cannot be moved to `observability` config mentioned above (where it clearly belongs) because of compatibility concerns? That's a pity.

Reply:
while we are here, I think we can to move it for all configs

Reply:
Yep, this is what I meant. However, AFAIU, we'd hit Protobuf incompatibility in this case; or is simply moving a config not a concern at this point? (I.e., not caring that it could be specified in the old location somewhere.)

Reply:
it makes it slightly harder to work with protobuf configs. 
Probably we can address it in the next step

Reply:
I've found that i have prometheus filed in non api.. 
but seems i have to revisit all Prometheus configs